library/src/scala/quoted/Toolbox.scala

```diff
@@ -4,9 +4,7 @@ import scala.annotation.implicitNotFound
 
 @implicitNotFound("Could not find implicit quoted.Toolbox.\n\nDefault toolbox can be instantiated with:\n  `implicit val toolbox: scala.quoted.Toolbox = scala.quoted.Toolbox.make(getClass.getClassLoader)`\n\n")
 trait Toolbox {
-  def run[T](expr: Expr[T]): T
-  def show[T](expr: Expr[T]): String
-  def show[T](tpe: Type[T]): String
+  def run[T](expr: QuoteContext => Expr[T]): T
 }
```

Comment:
Is it possible to merge the concept `Toolbox` with `QuoteContext`?

Reply:
Not really. The toolbox represents an instance of the compiler (including caches) while the QuoteContext corresponds to a single compilation run.